### PR TITLE
Goon chem fixes and reorganization

### LINKED ISSUE #7675: Stimpack doesn't work as advertised in the uplink
@Iamgoofball Well done, you caused a team of nuke ops to completely get shit on by tasers since it doesn't actually make you immune to stuns. :laughing: 

Also, there's literally no reason for this to exist, since adrenal implants are the same price and way easier to use.

### LINKED ISSUE #8493: Sterilizine and antihol recipes are incompatible.
The result is that chemists can't make sterilizine.

Antihol: ethanol, charcoal
Sterilizine: ethanol, charcoal, chlorine

### LINKED ISSUE #8500: Chlorine Triflouride foam causes placeholder tiles to appear when melting walls
Walls melted by Chlorine Triflouride foam leave behind a HONK tile, which I am pretty sure is a placeholder tile.

### LINKED ISSUE #8621: stimpacks are worthless
they don't work at all. seems they don't actually affect stun or is parallelize much and instead acts as mega heals.i blame goofball.how is syndiesauce weaker then meth.

### LINKED ISSUE #8632: Aranesp - Goofball pls.
Shit kills you dead in few seconds, is lethal at less than 5u and can be ghetto made using patients room medivend. Really has to be nerfed.

### LINKED ISSUE #8831: Cryosting does nothing.
To reproduce: 
- get stung
- check that you have frostoil in your reagents
- check your bodytemperature not changing at all.

### LINKED ISSUE #8849: Literally every on mix goon reagent is fucked
Since goof cannot remember things I tell him, here's a pr.

Every single goon chem that has a 'reaction on mix' creates, and leaves over the resultant reagent, rather then consuming it, where as tg 'react on mix' chems create just a reaction, leaving nothing left over.

This was partly addressed following the LDM smoke server crash thing, but chemicals such as Life still have this bug.

### LINKED ISSUE #8945: BLACK POWDER CAUSES INFINITY EXPLOSIONS AND CRASHES SERVERS
@Iamgoofball GOD FUCKING DAMNIT STOP CRASHING THE FUCKING SERVERS

### PR DESCRIPTION
**Pull request by phil235**
Reorganizing the reagents files. Goon chems are no longer in separate files. Moved grenade launcher and syringe gun from reagents module folder to gun folder.
The reagents module will now look like this:
- Chemistry-Colours.dm
- Chemistry-Holder.dm
- Chemistry-Machinery.dm
- Chemistry-Reagents.dm     <---- no reagents defined there, only procs.
- Chemistry-Recipes.dm        <---- no chemical reactions defined there, only procs.
- reagents_containers.dm
- reagent_dispensers.dm
- Chemistry-Reagents (Folder)
  - Blob, Drug, Medicine, Others, Pyrotechnics, Toxins (all .dm) (the reagents)
  - Consumable-Reagents (Folder)
- Chemistry-Recipes (Folder)
  - Drug, Medicine, Others, Pyrotechnics, Slime_extracts, Toxins (all .dm) (the chem reactions)
- reagent_containers (Folder)

ALSO:
- All reagents are back to being metabolized every tick. I adjusted the goon chem so their effects aren't changed too much.
- Removing some probability and randomness from goon chem effects to make it more predictable.
- Adding overdose to space drug.
- Making a drug reagent category.
- Changed check_ear_prot() for carbons: alien are now immune, and monkey checks their head for ear protection.
- Changed flashpowder flashing code and blackpowder explosion code to use already existing procs.
- Moved chemical_mob_spawn and goon_vortex procs to /datum/chemical_reaction/
- Fixes double overdose message for sugar.
- Remove the "if(!M) M = holder.my_atom" at the beginning of a lot of on_mob_life procs, you already can't call on_mob_life if M is null(Chemistry-Holder.dm).
- Moved random_color_list to datum/reagent/colorful_reagent
- Nerfed healing power of stimulant. Buffed ephedrine a bit.
- Fixes some goon chem on_reaction() proc leaving resulting chem behind when they shouldn't.
- Removes Life and Corgium reagents. They're now only a chemical reaction.
- Added a message for the ghost when someone is trying to revive its corpse with strange reagent. (like defib)
- Fixes Frost oil and other consumable reagents using current_cycle var.
- Fixes runtime where during metabolization a reagent removes another then the latter still calls on_mob_life().
- Fixes antihol only removing pure ethanol instead of all alcoholic reagents.
- Changed antihol recipe (now needs copper) to avoid conflict with sterilizine.
- Fixes broken floor without icon when burning wall with clf3, it now makes a plating instead of a tile. Also, clf3 has a chance to burn platings/tiles if it doesn't destroy it.
- Makes Nitroglycerin stabilizable with stabilizing agent and can explode when heated to 474K.
- Some other tiny fixes.

Fixes #8945.
Fixes #8831.
Fixes #8632
Fixes #8849
Fixes #8500
Fixes #8493 
Fixes #8621
Fixes #7675.

### CODE REVIEW

**Comment by hornygranny** [comment]
TYBP

**Comment by Menshin** [comment]
Great work :+1:

**Comment by AlexCrimson** [comment]
Did you make all the reagents in the Chem Dispenser list alphabetically? I feel like that is a super high priority fix.

**Comment by Fox-McCloud** [comment]
Goon's chems are meant to be processed every tick---it's just the life proc is meant to take place once every 2 seconds instead of ~1.5-1.8.

If you guys ever port the process scheduler, there won't be need to adjust reagent numbers.

**Comment by Dorsisdwarf** [comment]
Nitroglycerine should be an actual reagant because it's a real chemical that can actually be stabilised/

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 1]
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
...
+	id = "calomel"
+	description = "Quickly purges the body of all chemicals. If your health is above 20, toxin damage is dealt. When you hit 20 health or lower, the damage will cease."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+
+datum/reagent/medicine/calomel/on_mob_life(var/mob/living/M as mob)
+	for(var/datum/reagent/R in M.reagents.reagent_list)
+		if(R != src)
+			M.reagents.remove_reagent(R.id,2.5)
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Whoa whoa whoa hold the fuck up here

Revert this change! You nerfed calomel into the ground!

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 2]
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
...
+	..()
+	return
+
+datum/reagent/medicine/potass_iodide
+	name = "Potassium Iodide"
+	id = "potass_iodide"
+	description = "Reduces low radiation damage very effectively."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 2 * REAGENTS_METABOLISM
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Just get rid of REAGENTS_METABOLISM entirely, please. Unless it's equal to 1.

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 3]
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
...
+	id = "salbutamol"
+	description = "Quickly heals oxygen damage while slowing down suffocation. Great for stabilizing critical patients!"
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.25 * REAGENTS_METABOLISM
+
+datum/reagent/medicine/salbutamol/on_mob_life(var/mob/living/M as mob)
+	M.adjustOxyLoss(-3*REM)
+	if(M.losebreath >= 4)
+		M.losebreath -= 2
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: This needs to be -4.

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 4]
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
...
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 30
+	addiction_threshold = 25
+
+
+datum/reagent/medicine/morphine/on_mob_life(var/mob/living/M as mob)
+	M.status_flags |= IGNORESLOWDOWN
+	if(cycle_count >= 36)
+		M.sleeping += 3
+	cycle_count++
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Why did you remove my cycle_count variable from the base reagents thing? Please move var/cycle_count back up to all reagents and make it increase by 1 every time on_mob_life() is called.

File: code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm [hunk 5]
```diff
@@ -18,13 +57,6 @@
...
 
-/datum/reagent/clf3
-	name = "Chlorine Trifluoride"
-	id = "clf3"
-	description = "Makes a temporary 3x3 fireball when it comes into existence, so be careful when mixing. ClF3 applied to a surface burns things that wouldn't otherwise burn, sometimes through the very floors of the station and exposing it to the vacuum of space."
-	reagent_state = LIQUID
-	color = "#FF0000"
-	metabolization_rate = 4
 
 /datum/chemical_reaction/clf3
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm]: Why did you separate the recipes from the reagents?

I kept them in the same file as the reagents for easy access to tweaking all things about a reagent without having to flip between files.

File: code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm [hunk 6]
```diff
@@ -176,27 +141,12 @@
...
 	sleep(rand(50,100))
-	blackpowder_detonate(holder, created_volume)
-	return
-
-/datum/reagent/blackpowder/on_ex_act()
-	blackpowder_detonate(holder, volume)
-	return
+	var/location = get_turf(holder.my_atom)
+	var/datum/effect/effect/system/reagents_explosion/e = new()
+	e.set_up(round(created_volume/8, 1), location, 1, 4, message = 0)
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm]: Please make sure the explosion scales up to 300 units /exactly/.

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 7]
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
...
+	reagent_state = LIQUID
+	color = "#CF3600"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	toxpwr = 0
+
+datum/reagent/toxin/neurotoxin2/on_mob_life(var/mob/living/M as mob)
+	if(M.brainloss + M.toxloss <= 60)
+		M.adjustBrainLoss(1*REM)
+		M.adjustToxLoss(1*REM)
+	if(current_cycle == 54)
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Might wanna change this to 
if(current cycle >= 54)

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 8]
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
...
+	..()
+
+datum/reagent/toxin/cyanide
+	name = "Cyanide"
+	id = "cyanide"
+	description = "Deals toxin damage, alongside some oxygen loss. 8% chance of stun and some extra toxin damage."
+	reagent_state = LIQUID
+	color = "#CF3600"
+	metabolization_rate = 0.125 * REAGENTS_METABOLISM
+	toxpwr = 0.75
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Please don't put raw toxpwr on these. I don't want them doing raw toxin damage in some instances.

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 9]
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
...
+	name = "Initropidril"
+	id = "initropidril"
+	description = "Causes some toxin damage, 5% chances to cause stunning, suffocation, or immediate heart failure."
+	reagent_state = LIQUID
+	color = "#CF3600"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	toxpwr = 2.5
+
+datum/reagent/toxin/initropidril/on_mob_life(var/mob/living/M as mob)
+	if(prob(5))
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Revert this if(prob(5)) shit.

Revert it on all the chems you added it to. It made all the toxins damn near worthless.

Initro is one of the ones you spend TCs on, why did you nerf it into the ground?

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 10]
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
...
+	name = "Amanitin"
+	id = "amanitin"
+	description = "On the last second that it's in you, it hits you with a stack of toxin damage based on how long it's been in you. The more you use, the longer it takes before anything happens, but the harder it hits when it does."
+	reagent_state = LIQUID
+	color = "#CF3600"
+	toxpwr = 0
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+
+datum/reagent/toxin/amanitin/on_mob_delete(var/mob/living/M as mob)
+	M.adjustToxLoss(current_cycle*3*REM)
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Take the *REM out of this.

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 11]
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
...
+	name = "Cyanide"
+	id = "cyanide"
+	description = "Deals toxin damage, alongside some oxygen loss. 8% chance of stun and some extra toxin damage."
+	reagent_state = LIQUID
+	color = "#CF3600"
+	metabolization_rate = 0.125 * REAGENTS_METABOLISM
+	toxpwr = 0.75
+
+datum/reagent/toxin/cyanide/on_mob_life(var/mob/living/M as mob)
+	if(prob(5))
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Why is this prob(5)?

**Comment by Iamgoofball** [comment]
Overall, 7/10. Fix those marked issues and we're good to go.

**Comment by Iamgoofball** [comment]
Also, I intentionally had random_color_list global so that we could use it in the future for other things.

**Comment by optimumtact** [comment]
33 percent chance to stun was shit goof

**Comment by Iamgoofball** [comment]
Then lower the stun chance and don't wrecking ball the rest?

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 2 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Isn't it used to adjust the overall metabolization speed through config?

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 3 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: It was -4 and being called once every two life(). Now that it's called every life() the effect needs to be halved.

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 4 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: will do

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm (hunk 7 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: will do

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm (hunk 8 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: I didn't change anything, cynaide/on_mob_life() already had adjustToxLoss(0.75), I simply moved it to be dealt in toxin/on_mob_life()

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm (hunk 9 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: I don't understand, didn't you put that prob there? Also I didn't nerf that chemical, it was originally a prob(7) but since the chemical is now metabolized during every life() I reduced it to prob(5) so in fact I buffed it! The prob(33) that was in the proc was for the adjustToxLoss, all I did was remove the probably and randomness, instead of 33% of rand(5,25) once every two life we now get 2.5 every life() (0.33*0.5*(5+25)/2 = 2.5)

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm (hunk 10 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: REM was already there, it's used for nearly all reagent damage, plus it's equal to 1 by default.

File: code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm (hunk 5 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm]: Because the original chemicals already separated them, plus it makes smaller files.

File: code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm (hunk 6 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm]: I tried my best to stick to the numbers in the blackpowder explosion proc.

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm (hunk 8 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Oh, okay.

File: code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm (hunk 6 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm]: Alright.

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Last I checked it was a hard define. It's 0.6.

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm (hunk 9 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Ah.

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 2 above)
Comment (Cheridan) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: It's not a config, but it was added so that it's easy to adjust metabolization speed. It was originally added as a baystation compatibility update, since their chems metabolize more slowly.

**Comment by phil235** [comment]
Should be ready to go.

**Comment by phil235** [comment]
Good job Travis...

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Whoever set the bay metabolization rates had to have been insane. Some of them metabolize slower then black powder does.

Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: I did not simply multiply by REAGENTS_METABOLISM, the metabolization rates are unchanged, instead of reading "2" it reads "4 * REAGENTS_METABOLISM" for example.

Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: What? it says "metabolization_rate = 2 * REAGENTS_METABOLISM" on my end. That's a rate of 0.8.
